src/hotspot/share/runtime/javaThread.cpp

```diff
@@ -1170,7 +1171,7 @@ void JavaThread::set_is_VTMS_transition_disabler(bool val) {
 bool JavaThread::java_suspend() {
 #if INCLUDE_JVMTI
   // Suspending a JavaThread in VTMS transition or disabling VTMS transitions can cause deadlocks.
-  assert(!is_in_VTMS_transition(), "no suspend allowed in VTMS transition");
+  assert(!is_in_non_tmp_VTMS_transition(), "no suspend allowed in VTMS transition");
```

Comment:
IMHO, a non tmp VTMS transition should be considered a type of VTMS transition, so the assert check here does not really match the assert text. Also see my related comments below on naming and use of these flags and APIs.

Reply:
Thanks. Accepted.

Reply:
I thought I was pointing out a naming issue, but your change indicates that it was actually a bug, and it is ok to be in a tmp transition here. Is that correct?

Reply:
Chris, I believe, this comment is resolved. Please, let me know if you disagree.